source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -450,19 +411,11 @@ void MSETRequest::onChildResponse(Common::Redis::RespValuePtr&& value, uint32_t
 }
 
 void MSETRequest::recreate(Common::Redis::RespValue& request, uint32_t index) {
-  static const uint32_t SET_COMMAND_SUBSTRINGS = 3;
-  uint32_t num_values = SET_COMMAND_SUBSTRINGS;
-  std::vector<Common::Redis::RespValue> values(num_values);
-
-  for (uint32_t i = 0; i < num_values; i++) {
-    values[i].type(Common::Redis::RespType::BulkString);
-  }
-  values[--num_values].asString() = incoming_request_->asArray()[(index * 2) + 2].asString();
-  values[--num_values].asString() = incoming_request_->asArray()[(index * 2) + 1].asString();
-  values[--num_values].asString() = "set";
-
-  request.type(Common::Redis::RespType::Array);
-  request.asArray().swap(values);
+  Common::Redis::RespValue::CompositeArray single_set(
```

Comment:
Every place where you do magic math like this can you add more comments?

Reply:
Ok, I'll add a comment.  That said, all the recreate method will be remove in the next PR.